src/main/java/org/cloudfoundry/promregator/auth/OAuth2ServiceConfig.java

```diff
@@ -0,0 +1,87 @@
+package org.cloudfoundry.promregator.auth;
+
+import static com.sap.cloud.security.config.cf.CFConstants.CERTIFICATE;
+import static com.sap.cloud.security.config.cf.CFConstants.CLIENT_ID;
+import static com.sap.cloud.security.config.cf.CFConstants.CLIENT_SECRET;
+import static com.sap.cloud.security.config.cf.CFConstants.KEY;
+import static com.sap.cloud.security.config.cf.CFConstants.URL;
+import static com.sap.cloud.security.config.cf.CFConstants.XSUAA.CERT_URL;
+import static com.sap.cloud.security.config.cf.CFConstants.XSUAA.CREDENTIAL_TYPE;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.cloudfoundry.promregator.config.OAuth2XSUAAAuthenticationConfiguration;
+
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.config.Service;
+
+public class OAuth2ServiceConfig implements OAuth2ServiceConfiguration {
+
+	private final Map<String, String> properties = new HashMap<>();
+
+	public OAuth2ServiceConfig(OAuth2XSUAAAuthenticationConfiguration config) {
+		properties.put(CERTIFICATE, config.getClient_certificates());
+		properties.put(KEY, config.getClient_key());
+		properties.put(CLIENT_ID, config.getClient_id());
+		properties.put(CLIENT_SECRET, config.getClient_secret());
+		properties.put(URL, config.getUrl());
+		properties.put(CERT_URL, config.getCertUrl());
+		properties.put(CREDENTIAL_TYPE, config.getCredential_type());
```

Comment:
> why not giving it dedicated attributes

Did this intentionally. Reason: The interface which we implement here defines a `getProperty(./.)` method and uses it inside the default impl of method `getClientIdentity` also requiring the property keys `certificate` and `key`. Of course it would be possible to implement that also via reflection, but that seems to be overkill to me.